workers/default/xml/tasks.py

```diff
@@ -56,29 +28,47 @@ class GenerateXMLTask(ObjectTask):
     name = "generate_xml"
 
     def process_object(self, obj):
-        articles_meta = []
-        for part in obj.get_parts():
-            articles_meta.append({
-                **part.metadata.to_dict(),
-                'filepath': os.path.join(part.get_representation_dir(
-                                Object.INITIAL_REPRESENTATION),
-                            'merged.pdf')  # TODO filename may be different?
-            })
+        xml_type = self.get_param('xml_type')
+
+        if xml_type == 'ojs':
```

Comment:
Das hier ist wieder nicht so gut erweiterbar. Jedesmals, wenn wir ein neues XML-Format brauchen muss man wieder an den Code. Und wenn man sowieso wieder an den Code muss dann wäre die alte Lösung mit einem Task pro Format fast schöner als diese if-Wurst.

Besser wäre es, alle Variablen, die hier gesetzt werden als Task-Parameter nach außen zu führen und in der YAML zu setzen. Dann muss man nur die Schemata und Templates erstellen und die YAML anpassen, wenn man ein neues XML-Format braucht.